File: leetcode-cn/Offer06.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public int[] reversePrint(ListNode head) {
        ListNode node = head;
        int count = 0;
        while(node != null) {
            count++;
            node = node.next;
        }

        int[] res = new int[count];

        int index = count - 1;
        while(head != null) {
            res[index] = head.val;
            head = head.next;
            index--;
        }
        return res;
    }
}

File: leetcode-cn-go/offer06.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */
func reversePrint(head *ListNode) []int {
    node := head
    count := 0
    for node != nil {
        count++
        node = node.Next
    }

    res := make([]int, count)
    for i := count - 1; i >= 0; i-- {
        res[i] = head.Val
        head = head.Next
    }
    return res
}
